### PR TITLE
Refactoring non-idempotent fix with contextmanager a bit

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: pyteal/ast/router.py [hunk 1]
```diff
@@ -900,6 +901,13 @@ def _build_program(
             self.contract_construct(),
         )
 
+    @contextmanager
+    def __revert_side_effect(self):
```
Comment (tzaffi) [inline pyteal/ast/router.py:905]:
```suggestion
    def _cleaning_context(self):
```

nit: proposing slightly more unified naming

**Review by tzaffi — APPROVED** [review]
This is much better, thank you! (but a small nit suggesting name change)

File: pyteal/ast/router.py (hunk 1 above)
Comment (ahangsu) [reply pyteal/ast/router.py]: sure, that makes sense

Comment (tzaffi) [reply pyteal/ast/router.py]: Did you inted `_revert_side_effect` rather than `_cleaning_context`. Either is fine (even if I suggested the latter)

**Review by tzaffi — APPROVED** [review]

**Pull request merged**